drivers/gpio/gpio_davinci.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_davinci_gpio
+
+#include <errno.h>
+
+#include <zephyr/arch/common/sys_bitops.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/kernel.h>
+
+/* Helper Macros for GPIO */
+#define DEV_GPIO_CFG(dev)	\
+		((const struct gpio_davinci_config * const)((dev)->config))
+#define DEV_GPIO_CFG_BASE(dev) \
```

Comment:
@slpp95prashanth can we use DEVICE_MMIO_GET()/DEVICE_MMIO_NAMED_GET()  to get the base address? this driver should be generic and should work with other K3 platforms as well.